crates/cli-support/src/js/mod.rs

```diff
@@ -2831,11 +2981,24 @@ __wbg_set_wasm(wasm);"
                             self.typescript.push_str(";\n");
                         }
 
-                        self.export(name, &format!("function{}", code), Some(&js_docs))?;
+                        self.export(
+                            name,
+                            ExportJs::Function(&format!("function{}", code)),
+                            Some(&js_docs),
+                        )?;
                         self.globals.push('\n');
                     }
                     AuxExportKind::Constructor(class) => {
-                        let exported = require_class(&mut self.exported_classes, class);
+                        let exported = self.require_class_or_namespace(class);
+                        let exported = match exported {
+                            ClassOrNamespace::Class(class) => class,
+                            ClassOrNamespace::Namespace(_) => {
+                                bail!(
+                                    "constructor is not supported on `{}`. Enums do not support exported constructors.",
+                                    class
+                                );
```

Comment:
This has to happen in the proc-macro instead. In which case this becomes `unreachable!()`.
A UI test would be great as well.

Reply:
That will be difficult. We don't know whether a type is a string or an enum until the whole program is parsed.

Reply:
Yeah, that's bad ... I guess that explains why this was implemented how it was until now.

I guess the best we can do now is emit a warning in `wasm-bindgen`, but erroring out is not good. Not all library authors actually run `wasm-bindgen-cli` in their CI, they just `cargo build --target wasm32-unknown-unknown` and call it a day.

Reply:
Just remembered that this can actually be done via the `ClassMarker` (or w/e its called, on my phone rn).

Reply:
See my comment in https://github.com/rustwasm/wasm-bindgen/pull/4258#issuecomment-2477492640, this is definitely possible.